File: ch1/gugu.py
# 6. 파이썬으로 구구단 만들기
# for y in range(1, 10):
#     for x in range(1, 10):
#         print("{:3},".format(y * x), end="")
#     print("")

# 행의 마지막 값엔 쉼표(,) 표시하지 않기
for y in range(1, 10):
    a = ["{:3}".format(y * x) for x in range(1, 10)]
    print(",".join(a))

File: ch1/gugu.rs
// 6. 러스트로 구구단 만들기
// fn main(){
//     for y in 1..10 {
//         for x in 1..10 {
//             print!("{:3}", (y*x));
//             if x < 9 {
//                 print!(",")
//             }
//         }
//         println!("")
//     }
// }

// 마지막 행엔 쉼표(,) 제거하기
fn main(){
    for y in 1..10 {
        let s = (1..10)
                .map(|x| format!("{:3}", (x * y)))
                .collect::<Vec<String>>().join(",");
            println!("{}", s)
    }
}